teps/0133-configure-default-resolver.md

```diff
@@ -0,0 +1,164 @@
+---
+status: implementable
+title: Configure Default Resolver
+creation-date: '2023-03-03'
+last-updated: '2023-03-03'
+authors:
+- '@QuanZhang-William'
+- '@vdemeester'
+collaborators: []
+---
+
+# TEP-0133: Configure Default Resolver
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Use Cases](#use-cases)
+      - [Custome Resolver](#custom-resolver)
+      - [Common Remote Resource Storage](#common-remote-resource-storage)
+- [Proposal](#proposal)
+- [Design Details](#design-details)
+    - [Populate Default Resolver](#populate-default-resolver)
+    - [Validation Webhook](#validation-webhook)
+- [Design Evaluation](#design-evaluation)
+  - [Simplicity](#simplicity)
+  - [Flexibility](#flexibility)
+  - [Reusability](#reusability)
+- [Alternatives](#alternatives)
+- [References](#references)
+<!-- /toc -->
+
+## Summary
+This TEP proposes to support configurable default `Resolver` type to improve the simplicity of *Remote Resolution*. This TEP builds on prior work [TEP-0060].
+
+## Motivation
+Today, Tekton users must explicitly specify the `Resolver` type to use Remote Resolution, which brings verbosity when authoring the input yaml file.
+
+### Goals
+- Allow cluster operators to configure a default `Resolver` type when using Tekton Pipelines .
+- The default `Resolver` can either be a built-in type (`hub`, `git`, `cluster`, and `bundle`) or a [Custom Resolver].
+
+### Use Cases
+#### Custom Resolver
+As a company cluster operator, I may have my own [Custom Resolver] which is used by most of the cluster users when resolving remote resources. I may want to simplify the user experience with such default values at the authoring time.
+
+#### Common Remote Resource Storage
+As a company cluster operator, all of the company's `Tasks` and `Pipelines` are stored in the same GitHub repo. I may not want to enforce the company users to specify the `Resolver` type for every `TaskRun` and `PipelineRun` (see the [issue #5907]).
+
+## Proposal
+To support the configurable default `Resolver` type, we propose to introduce a new field `default-resolver-type` in the [config-defaults.yaml][config-defaults]. 
+
+With the `default-resolver-type` field configured, authors can create `TaskRun` and `PipelineRun` without setting the `resolver` field in the yaml input. 
+
+For example:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata:
+ name: demo-task-run
+spec:
+ taskRef:
+   #resolver: git
+   params:
+   - name: url
+     value: https://github.com/tektoncd/catalog.git
+   - name: revision
+     value: main
```

Comment:
can these two fields be defaulted as well? wondering because the use case mentions all resources being stored in one github repo, so maybe they'd want to be able to default url? and maybe also that it uses "main" revision by default?

Reply:
Good point, these values have configurable default values and are optional. I added more details to this section.